### PR TITLE
Expand Test Coverage of Ingest.py to 80%

### LINKED ISSUE #321: Expand Test Coverage in tests/switchmap_/server/db/ingest/test_ingest.py
Increase the test coverage of the file tests/switchmap_/server/db/ingest/test_ingest.py to as close to 100% as possible.

**Requirements**
1. [The current code coverage for the file can be found here](https://app.codecov.io/gh/PalisadoesFoundation/switchmap-ng/tree/develop/?displayType=list)
2. CodeRabbit must approve all your changes to the test file(s) and the file(s) under test
3. The PR will show a report for the code coverage for the file you have added. You can use that as a guide
4. No functionality must be removed to achieve this goal
5. All tests must be valid and truly test the code base
6. Take guidance on the formatting of the file from other tests/switchmap_/x/y/z/test_*.py files or others along the path or in adjacent directories
7. Fix test_ipport: The current test_ipport function returns prematurely. Investigate and fix this. If it does not correctly test relevant functionality, it should be removed.

### PR DESCRIPTION
**Pull request by Abhi-MS**
**What kind of change does this PR introduce?**

Improve the test coverage of ingest.py from 39% to 80%

**Issue Number:**

Fixes #321

**Snapshots/Videos:**

image

**If relevant, did you update the documentation?**

Not relevant

**Summary**

This PR introduces a comprehensive test suite for the ingest mood. The test file ensures the correctness of various functionalities related to database ingestion, data processing, and configuration handling. 

Key Features:
- Database Reset and Setup
- Unit Tests for Core Functions
- Validation of Helper Functions
- Configuration Handling
- Edge Case Handling
- Mock Data and Directories

**Does this PR introduce a breaking change?**
No

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass

**Other information**

Future Enhancements:
- Complete the placeholder tests (e.g., test_cleanup) to ensure full coverage.
- Add more edge case tests for complex scenarios.
- Refactor repetitive code in setup and teardown methods for better maintainability.
- Improve testing for process function
- Investigate and fix the following issues:
      - Test interference issue when passing valid argument in test_zone function (due to rest_db calling insert_arptable function)
      - Cleanup function not deleting event in test mode affecting cleanup and process functions

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/switchmap-ng/blob/main/CONTRIBUTING.md)?**
Yes

## Summary by CodeRabbit

- **Tests**
  - Added comprehensive unit tests for ingestion features, including file processing, database updates, configuration handling, and file management.
  - Improved test setup and cleanup for reliable test environments.
  - Enhanced coverage with new tests for zone and device processing, event setup, and argument handling.

### CODE REVIEW

File: tests/switchmap_/server/db/ingest/test_ingest.py [hunk 1]
```diff
@@ -168,50 +310,303 @@ def tearDown(cls):
...
+
+    def test_get_zone_existing_zone(self):
+        """Test the case when the zone already exists."""
+        # Call the _get_zone function with the event and file path
+        result = _get_zone(self.event_row, self.filepath)
+
+        # Assertions to verify the behavior
+        # Check that a ZoneData object is returned
+        self.assertIsInstance(result, ZoneData)
+
```
[the remark this answers is not in the document]
Comment (Abhi-MS) [reply tests/switchmap_/server/db/ingest/test_ingest.py:578]: Keeping only test_get_zone_existing_zone as test_get_zone_creates_new_zone involves creation of a new file
